src/IssueViz.Security.UnitTests/Taint/TaintIssueToIssueVisualizationConverterTests.cs

```diff
@@ -130,262 +138,144 @@ public void Convert_FromSonarQubeIssue_IssueVizConverterCalledWithCorrectParamet
+            .Convert(Arg.Is<ITaintIssue>(x => x.HighestSoftwareQualitySeverity == null && x.Severity == expectedSeverity));
+    }
+
+    [TestMethod]
+    public void StandardSeverity_InvalidSeverity_Converts()
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new StandardModeDetails((IssueSeverity)999, default));
+
+        var act = () => testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        act.Should().Throw<ArgumentException>();
+        issueVizConverter.DidNotReceiveWithAnyArgs().Convert(default, default);
+    }
+
+    [TestMethod]
+    public void MqrSeverity_EmptySeverities_Throws()
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new MQRModeDetails(default, []));
+
+        var act = () => testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        act.Should().Throw<ArgumentException>();
+        issueVizConverter.DidNotReceiveWithAnyArgs().Convert(default, default);
+    }
+
+    [DataTestMethod]
+    [DataRow(ImpactSeverity.BLOCKER, SoftwareQualitySeverity.Blocker)]
+    [DataRow(ImpactSeverity.HIGH, SoftwareQualitySeverity.High)]
+    [DataRow(ImpactSeverity.MEDIUM, SoftwareQualitySeverity.Medium)]
+    [DataRow(ImpactSeverity.LOW, SoftwareQualitySeverity.Low)]
+    [DataRow(ImpactSeverity.INFO, SoftwareQualitySeverity.Info)]
+    public void MqrSeverity_SingleSeverity_ReturnsIt(
+        ImpactSeverity slCoreSeverity,
+        SoftwareQualitySeverity expectedSeverity)
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new MQRModeDetails(default, [new ImpactDto(SoftwareQuality.SECURITY, slCoreSeverity)]));
+
+        testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        issueVizConverter.Received()
+            .Convert(Arg.Is<ITaintIssue>(x => x.HighestSoftwareQualitySeverity == expectedSeverity && x.Severity == null));
+    }
+
+    [TestMethod]
+    public void MqrSeverity_InvalidSeverity_Throws()
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new MQRModeDetails(default, [new ImpactDto(SoftwareQuality.SECURITY, (ImpactSeverity)999)]));
+
+        var act = () => testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        act.Should().Throw<ArgumentOutOfRangeException>();
+    }
+
+    [DataTestMethod]
+    [DataRow(new[] { ImpactSeverity.LOW, ImpactSeverity.LOW, ImpactSeverity.LOW }, SoftwareQualitySeverity.Low)]
+    [DataRow(new[] { ImpactSeverity.LOW, ImpactSeverity.LOW, ImpactSeverity.INFO }, SoftwareQualitySeverity.Low)]
+    [DataRow(new[] { ImpactSeverity.LOW, ImpactSeverity.MEDIUM, ImpactSeverity.LOW }, SoftwareQualitySeverity.Medium)]
+    [DataRow(new[] { ImpactSeverity.HIGH, ImpactSeverity.MEDIUM, ImpactSeverity.LOW }, SoftwareQualitySeverity.High)]
+    [DataRow(new[] { ImpactSeverity.MEDIUM, ImpactSeverity.BLOCKER, ImpactSeverity.HIGH }, SoftwareQualitySeverity.Blocker)]
+    public void MqrSeverity_MultipleSeverities_ReturnsHighest(ImpactSeverity[] slCoreSeverities, SoftwareQualitySeverity expectedSeverity)
```

Comment:
Maybe rename test to
_MultipleSeverities => _MultipleSoftwareQualitySeverities OR _MultipleImpacts

Reply:
MqrSeverity is already in the title